scanner/src/main/kotlin/experimental/ScannerWrapper.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.KnownProvenance
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+import org.ossreviewtoolkit.scanner.ScanResultsStorage
+import org.ossreviewtoolkit.scanner.ScannerCriteria
+
+/**
+ * An interface to wrap source code scanners.
```

Comment:
Nit: I wonder whether we should keep the wording a bit more generic / vague here, as we're planning to implement "fake" scanners that just extract / process files from binary artifacts / archives.

Reply:
So just scratch "source code"?

Reply:
Maybe. I'm also struggling a bit with the use of "wrap" in here, as IMO it's not the interface itself that wraps the scanner, but the implementation (eventually). You *could* also implement the `ScannerWrapper` interface with a "real" scanner written in Kotlin, that does not "wrap" any CLI / remote scanner.

So how about "The base interface for all types of scanners"?